### ISSUE TITLE
cupy.scipyx.ndimage.map_coordinates requires coordinates array to be flattened

### ISSUE THREAD

**Issue by Mirebeau** [cupy/cupy#3933]
Dear cupy team,

I noted an inconsistency between the cupyx and scipy of ndimage.map_coordinates : the former requires the coordinates array to be flattened.
Users can get around this bug using a few array reshape, but it shouldn't be too difficult to fix within CuPy either.

Thanks for the great library btw.

```console
import cupyx.scipy.ndimage
import scipy.ndimage
import cupy as cp
z = cp.ones((10,10))
x = cp.ones((2,10,10))
scipy.ndimage.map_coordinates(z.get(),x.get()) # OK with numpy
cupyx.scipy.ndimage.map_coordinates(z,x.reshape((2,-1))) # OK with flattened coordinates array
cupyx.scipy.ndimage.map_coordinates(z,x) # ! fails !
```

OS : Windows
CuPy : 7.8
Cuda : 10.2

**Comment by grlee77** [comment]
This was fixed by #2813, but it doesn't look like the fix was backported to the 7.x branch.

**Comment by Mirebeau** [comment]
OK, thanks.
